### PR TITLE
[HW11] Add `CreateTask`, `UpdateComment` mutations

### PR DESCRIPTION
**Pull request by DataJoshua**
### Summary

A brief description of the pull request.

### How it works

Screenshots/screencasts of the pull request introduced functionality.

### Test plan

List of steps to manually test introduced functionality:

- Go to http://localhost:3000/graphiql
- Make request using schema:
```graphql
  query {
    me {
      id
      name
    }
  }
```

### Deploy notes

Notes regarding deployment the contained body of work.
These should note any db migrations, services, scripts, etc.

### Checklist:

- [ ] I have performed a self-review of my own code
- [ ] My pull request is shorter than 500 lines

### CODE REVIEW

File: app/graphql/mutations/create_task.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+module Mutations
+  class CreateTask < BaseMutation
+    argument :input, Types::Inputs::CreateTaskInput, required: true
+
+    type Types::Payloads::CreateTaskPayload
+
+    def resolve(input:)
+      project = Project.find(input[:project_id])
```
Comment (w1nway) [inline app/graphql/mutations/create_task.rb:8]: This variant are shorter
```suggestion
      project = Project.find(input.project_id)
```

**Review by w1nway — COMMENTED** [review]
Good job, my friend

File: app/graphql/mutations/create_task.rb (hunk 1 above)
Comment (Raily01) [reply app/graphql/mutations/create_task.rb:8]: this variant IS shorter:)

Comment (DataJoshua) [reply app/graphql/mutations/create_task.rb:8]: lol, thank u ))

File: app/graphql/mutations/create_task.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+module Mutations
+  class CreateTask < BaseMutation
```
Comment (RasimKhusaenov) [inline app/graphql/mutations/create_task.rb:2]: Can you move this mutation to another PR please?

File: app/graphql/mutations/update_comment.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+    type Types::Payloads::UpdateCommentPayload
+
+    def resolve(input:)
+      comment = Comment.find(input.id)
+      result = Comments::Update.call(comment: comment, comment_params: input.to_h)
+      if result.success?
+        result.to_h
+      else
+        result.to_h.merge({ errors: prepared_params(result.error) })
```
Comment (RasimKhusaenov) [inline app/graphql/mutations/update_comment.rb:13]: What is `prepared_params`?

File: app/graphql/mutations/create_task.rb (hunk 2 above)
Comment (DataJoshua) [reply app/graphql/mutations/create_task.rb:2]: done

Comment (RasimKhusaenov) [reply app/graphql/mutations/create_task.rb:2]: @DataJoshua what about removing it from this PR? :)

File: app/graphql/mutations/create_task.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+    def resolve(input:)
+      project = Project.find(input.project_id)
+      result = Tasks::Create.call(project: project, task_params: input.to_h)
+
+      if result.success?
+        result.to_h.merge({ errors: [] })
+      else
+        result.to_h.merge({ errors: prepared_params(result.project) })
+      end
```
Comment (RasimKhusaenov) [inline app/graphql/mutations/create_task.rb:15]: 1. prepared_params?
2. result.project?

File: app/graphql/mutations/update_comment.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    type Types::Payloads::UpdateCommentPayload
+
+    def resolve(input:)
+      comment = Comment.find(input.comment_id)
+      result = Comments::Update.call(comment: comment, comment_params: input.to_h.except(:comment_id))
+      if result.success?
+        result.to_h
+      else
+        result.to_h.merge({ errors: formatted_errors(result.comment) })
+      end
```
Comment (RasimKhusaenov) [inline app/graphql/mutations/update_comment.rb:16]:
```suggestion
      result.to_h.merge(errors: formatted_errors(result.comment))
```

**Review by RasimKhusaenov — CHANGES_REQUESTED** [review]
